### ISSUE TITLE
uds-k3d not working on k3d 5.7.0+

### ISSUE THREAD

**Issue by mjnagel** [defenseunicorns/uds-k3d#88]
### Environment

k3d 5.7.0

### Issue

k3d [5.7.0](https://github.com/k3d-io/k3d/releases/tag/v5.7.0) introduced a change to core-dns that now leverages the coredns-custom configmap by default. When trying to deploy our uds-dev-stack this results in an error due to helm ownership mismatching.

### Severity/Priority

Low/Medium, downgrading to pre-5.7.0 does work.

### Additional Context

This should be kept up to date + tested in the uds-k3d CI (and likely documented as "current tested/validated version(s)).

**Comment by mjnagel** [comment]
Noting that this behavior appears to have been reverted in 5.7.1 so this may be a non-issue in that version+. Still worth evaluating testing/documenting version support!

**Comment by unicorn-kp** [comment]
User reports they were on v5.7.1. Reverting to v5.6.3 allowed them to continue with the deployment.

**Comment by mjnagel** [comment]
@unicorn-kp in testing I'm unable to reproduce the issue on 5.7.1. I did encounter this problem on 5.7.0. Let me know if you are able to reproduce and we can reopen but I believe this issue is tied to upstream changes and seems to have been reverted.

Also noting that our docs do note the pre-req of k3d and [link the versions](https://github.com/defenseunicorns/uds-k3d?tab=readme-ov-file#prerequisites) from the uds-common repo. Currently our CI is testing with 5.7.1 and just confirmed latest CI runs look OK on #92.
